examples/hillas_analysis.py

```diff
@@ -0,0 +1,73 @@
+'''
+This example shows how you should use ctapipe.core.tool.Tool
+for your analysis as example for a simple Hillas based analysis.
+'''
+from ctapipe.core.tool import Tool
+from ctapipe.core.traits import Unicode, Dict, List
+from ctapipe.utils import get_dataset
+from ctapipe.io.eventsourcefactory import EventSourceFactory
+from ctapipe.calib import CameraCalibrator
+from ctapipe.image.hillas import hillas_parameters
+from ctapipe.image.cleaning import tailcuts_clean
+
+import astropy.units as u
+
+import json
+
+
+class HillasAnalysis(Tool):
+    # class members are options, that can be set via commandline or config file
+
+    # inputfile, if None we will use the gamma test file
+    inputfile = Unicode(None, allow_none=True).tag(config=True)
+
+    # outputfile
+    outputfile = Unicode('', allow_none=False).tag(config=True)
+
+    classes = List([CameraCalibrator, EventSourceFactory])
+
+    aliases = Dict({'outputfile': 'HillasAnalysis.outputfile'})
+
+    def setup(self):
+        print('inputfile', self.inputfile)
+        print('outputfile', self.outputfile)
+        self.event_source = EventSourceFactory.produce(
+            tool=self, config=self.config,
+            input_url=self.inputfile or get_dataset('gamma_test.simtel.gz')
+        )
+        self.calibrator = CameraCalibrator(
+            tool=self,
+            config=self.config,
+            event_source=self.event_source
```

Comment:
This should be "eventsource" instead of "event_source"

Reply:
I have to say I find this most surprising. Why are there even two parameters with similar names that behave so differently?  The `event_source` kwarg is not documented. Maybe we only allow one of these?

Reply:
Its because of kwargs. "event_source" does nothing, but doesn't throw an error when its used. Not sure how to better handle that, the kwargs is required as part of the traitlet functionality. 

We could just explicitly throw an error if someone attempts to use "event_source", but thats one specific fix for what is a wider (potential) problem.

Reply:
So that the traitlets can be configured when initialising the class

Reply:
kwargs is passed to `super().__init__`, `super()` of `CameraCalibrator` is `Component`. Which possible kwargs are we wanting to hand over to Component?

Reply:
any of the CameraR1Calibrator (and its subclasses) traitlets. Currently the only ones that exist are the ones in TargetIOR1Calibrator (pedestal_path, tf_path...). But this is not the only class that adheres to this pattern, every class that is a component uses kwargs in this way so that the traitlets are correctly configured.

Reply:
There are three ways traitlets can be set: passing a config file to the executable, setting them at the cmd line as arguments to the executable, or passing it as an argument in the initialisation of the class